stdlib/asm/crypto/fri/helper.masm

```diff
@@ -3,53 +3,68 @@ use.std::crypto::stark::random_coin
 use.std::crypto::stark::utils
 use.std::crypto::stark::constants
 
-
 #! Compute the number of FRI layers given log2 of the size of LDE domain. It also computes the
 #! LDE domain generator and, from it, the trace generator and store these for later use.
 #!
 #! Input: [...]
 #! Output: [num_fri_layers, ...]
-#! Cycles: 126
+#! Cycles: 123
 export.generate_fri_parameters
-    # Load  FRI verifier data:
-    # [lde_size, log(lde_size), 0, 0]
-    push.0.0.0.0
-    exec.constants::lde_size_ptr mem_loadw
+    # Load FRI verifier data
+    padw exec.constants::lde_size_ptr mem_loadw
+    #=> [lde_size, log(lde_size), 0, 0, ...] (6 cycles)
 
     # Compute lde_domain generator
     dup.1
     exec.utils::compute_lde_generator
-    #=> [g, lde_size, log2(lde_size), 0, ...]
     movdn.2
-    #=> [lde_size, log2(lde_size), g, 0, ...]
+    #=> [lde_size, log2(lde_size), lde_g, 0, 0, ...] (65 cycles)
 
     # Store in `TMP5` in order to use it for fri layer loading
     exec.constants::tmp5 mem_storew
 
     # Compute [gz1, gz0, z1, z0] using domain generator
     # TODO: move to somewhere else
-    push.0.0.0.0
+    # ---------------------------------------------------------------------------------------------
+
+    # load z from memory 
+    padw
     exec.constants::z_ptr mem_loadw
-    drop drop
+    #=> [(z1, z0)^n, z1, z0, lde_size, log2(lde_size), lde_g, 0, 0, ...] (6 cycles)
+
+    # prepare stack
+    drop
+    drop
     dup.1
     dup.1
     dup.6
+    #=> [lde_g, z1, z0, z1, z0, lde_size, log2(lde_size), lde_g, 0, 0, ...] (5 cycles)
 
-    # Compute trace_domain_generator
+    # Compute trace generator `trace_g` = `lde_g^blowup_factor`
     repeat.3
```

Comment:
This is an instance that would benefit from https://github.com/0xPolygonMiden/miden-vm/issues/968

Reply:
I think this as well as `compute_lde_generator` should be deprecated with the work you have done on the AirScript side, right?

Reply:
I suppose you're talking about the binary search? I closed that PR since the code was complicated https://github.com/0xPolygonMiden/air-script/pull/320#issuecomment-1606963893 .

Instead I'm going to save this value to memory and the `AirScript` codegen will load from the memory.